ElectromagneticModels/coupledElectromagneticModels/coupledCurrentDensity/coupledCurrentDensityFvPatchVectorField.C

```diff
@@ -0,0 +1,232 @@
+        valueFraction() = symmTensorField("valueFraction", dict, p.size());
+    }
+    else
+    {
+        //valueFraction() is symmTensorField:
+        //"xx", "xy", "xz",
+        //      "yy", "yz",
+        //            "zz"
+        valueFraction() = sqr(patch().nf());//fraction set to normal direction
+        // Thus, refValue() is normalCurrent
+        // Act as slip condition by default
+        refValue() = Zero;
+        //refGrad not used
+        refGrad() = Zero;
+    }
+    //evaluate() is called after all solvers have been constructed
+}
+
+
+Foam::coupledCurrentDensityFvPatchVectorField::
+coupledCurrentDensityFvPatchVectorField
+(
+    const coupledCurrentDensityFvPatchVectorField& psf,
+    const fvPatch& p,
+    const DimensionedField<vector, volMesh>& iF,
+    const fvPatchFieldMapper& mapper
+)
+:
+    directionMixedFvPatchVectorField(psf, p, iF, mapper)
+{}
+
+
+Foam::coupledCurrentDensityFvPatchVectorField::
+coupledCurrentDensityFvPatchVectorField
+(
+    const coupledCurrentDensityFvPatchVectorField& psf,
+    const DimensionedField<vector, volMesh>& iF
+)
+:
+    directionMixedFvPatchVectorField(psf, iF)
+{}
+
+
+// * * * * * * * * * * * * * * * Member Functions  * * * * * * * * * * * * * //
+
+void Foam::coupledCurrentDensityFvPatchVectorField::updateCoeffs()
+{
+    if (updated())
+    {
+        return;
+    }
+    
+    if (!coupled_)
+    {
+        //Act as slip condition by default
+        //Use default coefficients
+        return;
+    }
+    // Since we're inside initEvaluate/evaluate there might be processor
+    // comms underway. Change the tag we use.
+    int oldTag = UPstream::msgType();
+    UPstream::msgType() = oldTag + 1;
+
+    // Get the coupling information from the mappedPatchBase
+    const mappedPatchBase& mpp = mappedPatchBase::getMap(patch().patch());
+    const label patchiNbr = mpp.nbrPolyPatch().index();
+    const fvPatch& patchNbr =
+        refCast<const fvMesh>(mpp.nbrMesh()).boundary()[patchiNbr];
+
+    const fvPatchScalarField& ePotpNbr =
+        patchNbr.lookupPatchField<volScalarField, scalar>(ePotName_);
+
+    if (!isA<coupledElectricPotentialFvPatchScalarField>(ePotpNbr))
+    {
+        FatalErrorInFunction
+            << "Patch field for " << internalField().name() << " on "
+            << this->patch().name() << " is of type "
+            << coupledCurrentDensityFvPatchVectorField::typeName
+            << endl << "The neighbouring patch field " << ePotName_
+            << " on " << patchNbr.name() << " is required to be "
+            << coupledElectricPotentialFvPatchScalarField::typeName
+            << ", but is currently of type " << ePotpNbr.type()
+            << exit(FatalError);
+    }
+
+    const coupledElectricPotentialFvPatchScalarField& coupledPotentialNbr =
+        refCast<const coupledElectricPotentialFvPatchScalarField>(ePotpNbr);
+
+    tmp<scalarField> sigma;
+    // Get patch values
+    getThis(sigma);
```

Comment:
is this function name clear? I would assume getThis returs this object, which does not really make sense. Is there specific reason to call it that way? Also get... usually tells that it returns something.